### PR TITLE
Added possibility to show and edit vCard Geo attributes

### PR DESCRIPTION
**Pull request by tacruc**
Signed-off-by: Arne Hamann <kontakt+github@arne.email>

### CODE REVIEW

File: src/components/ContactDetails/ContactDetailsAddNewProp.vue [hunk 1]
```diff
@@ -101,10 +102,16 @@ export default {
 		 * @param {string} data.id the id of the property. e.g fn
 		 */
 		addProp({ id }) {
-			let defaultData = this.properties[id].defaultValue
-			let property = this.contact.vCard.addPropertyWithValue(id, defaultData ? defaultData.value : '')
-			if (defaultData && defaultData.type) {
-				property.setParameter('type', defaultData.type)
+			if (id === 'geo') {
```
Comment (skjnldsv) [inline src/components/ContactDetails/ContactDetailsAddNewProp.vue]:
```suggestion
			if (this.properties[id] && this.properties[id].defaultjCal) {
```

File: src/components/ContactDetails/ContactDetailsAddNewProp.vue [hunk 2]
```diff
@@ -101,10 +102,16 @@ export default {
...
+				let defaultjCal = this.properties[id].defaultjCal
+				let property = new ICAL.Property(['geo', {}, defaultjCal[1], defaultjCal[2]])
```
Comment (skjnldsv) [inline src/components/ContactDetails/ContactDetailsAddNewProp.vue]:
```suggestion
				let property = new ICAL.Property([id, ...defaultjCal])
```

File: src/models/rfcProps.js [hunk 3]
```diff
@@ -55,6 +55,12 @@ const properties = {
 		readableName: t('contacts', 'Website'),
 		icon: 'icon-public'
 	},
+	geo: {
+		multiple: true,
+		readableName: t('contacts', 'Location'),
+		icon: 'icon-address',
+		defaultjCal: [{}, 'URI', 'geo:90.000,0.000']
```
Comment (skjnldsv) [inline src/models/rfcProps.js]: Argh, I just realised it is not valid in vcard3.0.
Let's allow to specify multiple versions on a defaultjCal?
```suggestion
		defaultjCal: {
			'3.0': [{}, 'FLOAT', '90.000;0.000'],
			'4.0': [{}, 'URI', 'geo:90.000,0.000']
		}
```

File: src/components/ContactDetails/ContactDetailsAddNewProp.vue [hunk 4]
```diff
@@ -101,10 +102,16 @@ export default {
 		 * @param {string} data.id the id of the property. e.g fn
 		 */
 		addProp({ id }) {
-			let defaultData = this.properties[id].defaultValue
-			let property = this.contact.vCard.addPropertyWithValue(id, defaultData ? defaultData.value : '')
-			if (defaultData && defaultData.type) {
-				property.setParameter('type', defaultData.type)
+            if (this.properties[id] && this.properties[id].defaultjCal) {
```
Comment (skjnldsv) [inline src/components/ContactDetails/ContactDetailsAddNewProp.vue]:
```suggestion
            if (this.properties[id] && this.properties[id].defaultjCal
            	&& this.properties[id].defaultjCal[this.contact.version]) {
```

File: src/components/ContactDetails/ContactDetailsAddNewProp.vue [hunk 5]
```diff
@@ -101,10 +102,16 @@ export default {
...
+				let defaultjCal = this.properties[id].defaultjCal
```
Comment (skjnldsv) [inline src/components/ContactDetails/ContactDetailsAddNewProp.vue]:
```suggestion
				let defaultjCal = this.properties[id].defaultjCal[this.contact.version]
```

**Comment by skjnldsv** [comment]
@tacruc so what do you think, good enough?
Is it working properly on your side and fitting your needs? :)

**Comment by pieter-groeneweg** [comment]
I see this is under review (label "3. to review" and "enhancement") for milestone 3.1.4...
3.1.4 is completed as is 3.1.6 which I have installed. 
Out of curiousity... when can we expect GEO attributes in the contacts?

**Pull request merged**